Java/core/src/main/java/com/amazon/randomcutforest/IStateCoordinator.java

```diff
@@ -13,20 +13,21 @@
  * permissions and limitations under the License.
  */
 
-package com.amazon.randomcutforest.executor;
+package com.amazon.randomcutforest;
 
 import java.util.List;
 
+import com.amazon.randomcutforest.executor.UpdateResult;
+import com.amazon.randomcutforest.store.IPointStore;
+
 /**
- * An IUpdateCoordinator is used in conjunction with a family of IUpdatable
- * instances. The coordinator transforms the input point into the form expected
- * by the updatable models, and processes the list of deleted points if needed.
- * An IUpdateCoordinator can be used to manage shared state.
+ * An IStateCoordinator coordinates the updates as well as the translation of
+ * the internal representation to explicit points during inference.
  *
  * @param <PointReference> An internal point representation.
  * @param <Point>          Explicit point type
  */
-public interface IUpdateCoordinator<PointReference, Point> {
+public interface IStateCoordinator<PointReference, Point> {
```

Comment:
I'd like to revisit this name change. This interface defines methods  that are invoked before and after individual trees are updated, and it allows the Update Executor to sequence operations in a way that keeps the global state consistent. So I think "Update Coordinator" is not a bad name for this interface, even if we add the `getStore` method. What other operations are we thinking of using this interface for that would make the more general name "State Coordinator" a better fit?

Reply:
The main explicit task of the coordinator at the moment is to facilitate updates and that updates are not leaving the tree in an unusable "state".  However the way it is achieving this is by ensuring coordination between a sampler and a tree. However the point store is used in inference as well -- the trees only have a read only interface. One option is to make the point store forest level and have some glue (but any such glue would look this class) or have a state coordinator. Any "change of state" operation would need the point store and coordination (change the sample size parameter, change/reduce the shingle size parameter -- both of which has been asks).  Each such feature would need more coordination between samplers and trees. "Update" may not be the full description of the coordination required. But names other than StateCoordinator is welcome. PointCooordinator? We could also just call it Coordinator. From a constructor perspective, it makes sense to pass: trees, samplers, executioner's. context and a coordinator.